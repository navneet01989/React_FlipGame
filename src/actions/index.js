export const UPDATING_LIST_DATA = 'UPDATING_LIST_DATA';
export const UPDATE_COUNTER = 'UPDATE_COUNTER';
export const RECORD_POSITION = 'RECORD_POSITION';
export const UPDATE_PAIRS_DONE = 'UPDATE_PAIRS_DONE';

export const updatePairs = data => ({
    type: UPDATE_PAIRS_DONE,
    data: data
});

export const setListData = data => ({
    type: UPDATING_LIST_DATA,
    data: data
});

export const updateCounter = data => ({
    type: UPDATE_COUNTER,
    data: data
});

export const recordPosition = data => ({
    type: RECORD_POSITION,
    data: data
});

