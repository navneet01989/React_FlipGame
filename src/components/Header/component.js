import React from 'react';
import {
    Text,
    View,
    TouchableOpacity,
    StyleSheet
  } from 'react-native';
import { connect } from 'react-redux';
import { setListData, updateCounter } from '../../actions';
import { CARD_PAIRS_VALUE } from '../../../App';
import { populateList } from '../../utils';

const Header = ({dispatch, counter}) => {
    const onResetClick = () => {
      dispatch(updateCounter(0));
      dispatch(setListData(populateList(CARD_PAIRS_VALUE)));
    }
    return (
        <View style={styles.header}>
          <TouchableOpacity testID="resetBtn" onPress={onResetClick}><Text style={styles.reset}>Reset</Text></TouchableOpacity>
          <Text testID="counterText" style={styles.counterItem}>{`Steps ${counter}`}</Text>
        </View>
    )
}

const styles = StyleSheet.create({
    header: {
      flexDirection: 'row',
    },
    reset: {
      fontSize: 25,
      textAlign: 'left',
      flex:1,
      paddingLeft: 10,
      paddingRight:10,
      backgroundColor: 'cyan',
      color: 'black'
    },
    counterItem: {
      fontSize: 25,
      textAlign: 'right',
      flex:1,
      paddingLeft: 10,
      paddingRight:10,
    }
  });

  const mapStateToProps = state => ({
    counter: state.ListDataReducer.counter
  });

  export default connect(mapStateToProps)(Header);
