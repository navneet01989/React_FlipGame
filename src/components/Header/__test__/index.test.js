import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Header from '../../../../src/components/Header';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { UPDATE_COUNTER } from '../../../actions';
const mockStore = configureStore([]);

describe('Header snapshot', () => {
  let wrapper
  let store;
  beforeEach(() => {
    store = mockStore({
      ListDataReducer: {counter: 0}
    });
    wrapper = renderer.create(
      <Provider store={store}>
        <Header/>
      </Provider>
    );
  });
  afterEach(function() {
    store.clearActions()
  })

  test('onResetClick', () => {
    const button = wrapper.root.findByProps({ testID: 'resetBtn' });
    const stepsCounter = wrapper.root.findByProps({ testID: 'counterText' });
    expect(stepsCounter.props.children).toBe('Steps 0');
    act(() => {
      button.props.onPress();
    });
    expect(store.getActions()[0].type).toBe(UPDATE_COUNTER);  
  });
});