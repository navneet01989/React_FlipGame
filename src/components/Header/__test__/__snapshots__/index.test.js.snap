// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="resetBtn"
  >
    <Text
      style={
        Object {
          "backgroundColor": "cyan",
          "color": "black",
          "flex": 1,
          "fontSize": 25,
          "paddingLeft": 10,
          "paddingRight": 10,
          "textAlign": "left",
        }
      }
    >
      Reset
    </Text>
  </View>
  <Text
    style={
      Object {
        "flex": 1,
        "fontSize": 25,
        "paddingLeft": 10,
        "paddingRight": 10,
        "textAlign": "right",
      }
    }
    testID="counterText"
  >
    Steps 0
  </Text>
</View>
`;
